plugins/nominal-connection-checker/src/hierarchy_validation.js

```diff
@@ -228,6 +229,48 @@ function checkSupersDefined(hierarchyDef) {
   }
 }
 
+/**
+ * Checks the hierarchy def for any types which include the same type multiple
+ * times in their fulfills array.
+ * @param {!Object} hierarchyDef The definition of the type hierarchy.
+ */
+function checkDupeSupers(hierarchyDef) {
+  const errorMsg = 'The type %s fulfills the type %s multiple times.';
+
+  for (const type of Object.keys(hierarchyDef)) {
+    const typeInfo = hierarchyDef[type];
+    if (!typeInfo.fulfills || !Array.isArray(typeInfo.fulfills)) {
+      continue;
+    }
+    const superOccurrences = new Map();
+    const superNames = new Map();
+    for (const superType of typeInfo.fulfills) {
+      try {
+        const superName = parseType(superType, false).name;
+        const lowerCaseName = superName.toLowerCase();
+        if (!superNames.has(lowerCaseName)) {
+          superNames.set(lowerCaseName, superName);
+        }
+        if (!superOccurrences.has(lowerCaseName)) {
+          superOccurrences.set(lowerCaseName, 1);
+        } else {
+          superOccurrences.set(
+              lowerCaseName, superOccurrences.get(lowerCaseName) + 1);
+        }
+      } catch (e) {
+        if (!(e instanceof TypeParseError)) {
+          throw e;
+        } // Otherwise it will have been handled by our specific check.
+      }
+    }
+    for (const [lowerCaseName, occurences] of superOccurrences) {
+      if (occurences > 1) {
+        console.error(errorMsg, type, superNames.get(lowerCaseName));
```

Comment:
Idea (feel free to leave as is):
If you aren't actually using the count of occurrences in the error message, you could change your algorithm from a map to using sets.
Instead of the map `superOccurences`, you use the first set to store the supers that you encounter in fulfills and a second set for any multiples (if you find a super that's already in the first set) you encounter. Then the last loop will just go through that second set.